### ISSUE TITLE
Ruby 2.6

### ISSUE THREAD

**Issue by k2m30** [denofevil/debase#67]
Just switched to Ruby 2.6.0 from 2.5 on Mac OS

`gem install debase --pre`                                                                                                                      
Building native extensions. This could take a while...
ERROR:  Error installing debase:
	ERROR: Failed to build gem native extension.

    current directory: /Users/user/.rvm/gems/ruby-2.6.0/gems/debase-0.2.3.beta2/ext
/Users/user/.rvm/rubies/ruby-2.6.0/bin/ruby -I /Users/user/.rvm/rubies/ruby-2.6.0/lib/ruby/2.6.0 -r ./siteconf20181231-1553-2fb6q3.rb extconf.rb
checking for vm_core.h... yes
checking for iseq.h... yes
checking for version.h... yes
checking for vm_core.h... yes
checking for vm_insnhelper.h... yes
checking for vm_core.h... yes
checking for method.h... yes
creating Makefile

current directory: /Users/user/.rvm/gems/ruby-2.6.0/gems/debase-0.2.3.beta2/ext
make "DESTDIR=" clean

current directory: /Users/user/.rvm/gems/ruby-2.6.0/gems/debase-0.2.3.beta2/ext
make "DESTDIR="
compiling breakpoint.c
In file included from breakpoint.c:1:
In file included from ./debase_internals.h:5:
In file included from /Users/user/.rvm/src/ruby-2.6.0/vm_core.h:75:
In file included from /Users/user/.rvm/src/ruby-2.6.0/method.h:14:
/Users/user/.rvm/src/ruby-2.6.0/internal.h:125:10: warning: '__msan_allocated_memory' macro redefined [-Wmacro-redefined]
# define __msan_allocated_memory(x, y)
         ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.0.0/include/sanitizer/msan_interface.h:113:9: note: previous definition is here
#define __msan_allocated_memory(data, size)
        ^
In file included from breakpoint.c:1:
In file included from ./debase_internals.h:5:
In file included from /Users/user/.rvm/src/ruby-2.6.0/vm_core.h:75:
In file included from /Users/user/.rvm/src/ruby-2.6.0/method.h:14:
/Users/user/.rvm/src/ruby-2.6.0/internal.h:127:10: warning: '__msan_unpoison' macro redefined [-Wmacro-redefined]
# define __msan_unpoison(x, y)
         ^
/Library/Developer/CommandLineTools/usr/lib/clang/9.0.0/include/sanitizer/msan_interface.h:112:9: note: previous definition is here
#define __msan_unpoison(a, size)
        ^
In file included from breakpoint.c:1:
In file included from ./debase_internals.h:8:
/Users/user/.rvm/src/ruby-2.6.0/vm_insnhelper.h:43:16: error: use of undeclared identifier 'FALSE'
        return FALSE;
               ^
/Users/user/.rvm/src/ruby-2.6.0/vm_insnhelper.h:46:36: error: use of undeclared identifier 'FALSE'
        return RBASIC_CLASS(obj) ? FALSE : TRUE;
                                   ^
/Users/user/.rvm/src/ruby-2.6.0/vm_insnhelper.h:46:44: error: use of undeclared identifier 'TRUE'
        return RBASIC_CLASS(obj) ? FALSE : TRUE;
                                           ^
2 warnings and 3 errors generated.
make: *** [breakpoint.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/user/.rvm/gems/ruby-2.6.0/gems/debase-0.2.3.beta2 for inspection.
Results logged to /Users/user/.rvm/gems/ruby-2.6.0/extensions/x86_64-darwin-16/2.6.0/debase-0.2.3.beta2/gem_make.out

**Comment by rockorequin** [comment]
The error with "FALSE not defined" seems to be in the debase-ruby_core_source gem v0.10.4. I edited this file:

`/usr/local/rvm/gems/ruby-2.6.0/gems/debase-ruby_core_source-0.10.4/lib/debase/ruby_core_source/ruby-2.6.0-p0/vm_insnhelper.h`

and added the line: 

`#include "regenc.h"`

before this line: 

`RUBY_SYMBOL_EXPORT_BEGIN`

and this fixes the problem building debase v0.2.3.beta2 for me.

**Comment by valich** [comment]
I could not find any sensible way to compile `vm_insnhelper.h` from 2.6. However, it seems that it's not really needed for debase, I've cleaned up the includes, now it should work. Please check out `0.2.3.beta3`.

**Comment by Ingu** [comment]
On windows i'm still getting
```
Fetching debase 0.2.3.beta3 (was 0.2.2)
Installing debase 0.2.3.beta3 (was 0.2.2) with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
C:/Ruby26-x64/lib/ruby/gems/2.6.0/gems/debase-0.2.3.beta3/ext
C:/Ruby26-x64/bin/ruby.exe -I C:/Ruby26-x64/lib/ruby/2.6.0 -r
./siteconf20190115-12680-1s6z5xq.rb extconf.rb
checking for vm_core.h... no
checking for vm_core.h... no
checking for vm_core.h... yes
checking for iseq.h... yes
checking for version.h... yes
checking for vm_core.h... yes
checking for vm_insnhelper.h... yes
checking for vm_core.h... yes
checking for method.h... yes
creating Makefile

current directory: C:/Ruby26-x64/lib/ruby/gems/2.6.0/gems/debase-0.2.3.beta3/ext
make "DESTDIR=" clean

current directory: C:/Ruby26-x64/lib/ruby/gems/2.6.0/gems/debase-0.2.3.beta3/ext
make "DESTDIR="
generating debase_internals-x64-mingw32.def
compiling breakpoint.c
compiling context.c
compiling debase_internals.c
debase_internals.c: In function 'my_iseqw_check':
debase_internals.c:647:13: warning: implicit declaration of function
'ibf_load_iseq_complete'; did you mean 'rb_iseq_compile'?
[-Wimplicit-function-declaration]
             ibf_load_iseq_complete(iseq);
             ^~~~~~~~~~~~~~~~~~~~~~
             rb_iseq_compile
debase_internals.c: In function 'Debase_set_trace_flag_to_iseq':
debase_internals.c:659:31: warning: initialization discards 'const' qualifier
from pointer target type [-Wdiscarded-qualifiers]
             rb_iseq_t *iseq = my_iseqw_check(rb_iseq);
                               ^~~~~~~~~~~~~~
debase_internals.c:660:13: warning: implicit declaration of function
'rb_iseq_trace_set'; did you mean 'rb_iseq_pathobj_set'?
[-Wimplicit-function-declaration]
             rb_iseq_trace_set(iseq, RUBY_EVENT_TRACEPOINT_ALL);
             ^~~~~~~~~~~~~~~~~
             rb_iseq_pathobj_set
debase_internals.c: In function 'Debase_unset_trace_flags':
debase_internals.c:667:31: warning: initialization discards 'const' qualifier
from pointer target type [-Wdiscarded-qualifiers]
             rb_iseq_t *iseq = my_iseqw_check(rb_iseq);
                               ^~~~~~~~~~~~~~
debase_internals.c: In function 'Init_debase_internals':
debase_internals.c:724:64: warning: passing argument 3 of
'rb_define_module_function' from incompatible pointer type
[-Wincompatible-pointer-types]
rb_define_module_function(mDebase, "set_trace_flag_to_iseq",
Debase_set_trace_flag_to_iseq, 1);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/Ruby26-x64/include/ruby-2.6.0/ruby.h:33,
                 from ./debase_internals.h:4,
                 from debase_internals.c:1:
C:/Ruby26-x64/include/ruby-2.6.0/ruby/ruby.h:1787:50: note: expected 'VALUE
(*)()' {aka 'long long unsigned int (*)()'} but argument is of type 'void
(*)(VALUE,  VALUE)' {aka 'void (*)(long long unsigned int,  long long unsigned
int)'}
 void rb_define_module_function(VALUE,const char*,VALUE(*)(ANYARGS),int);
                                                  ^~~~~~~~~~~~~~~~~
debase_internals.c:727:58: warning: passing argument 3 of
'rb_define_module_function' from incompatible pointer type
[-Wincompatible-pointer-types]
rb_define_module_function(mDebase, "unset_iseq_flags",
Debase_unset_trace_flags, 1);
^~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/Ruby26-x64/include/ruby-2.6.0/ruby.h:33,
                 from ./debase_internals.h:4,
                 from debase_internals.c:1:
C:/Ruby26-x64/include/ruby-2.6.0/ruby/ruby.h:1787:50: note: expected 'VALUE
(*)()' {aka 'long long unsigned int (*)()'} but argument is of type 'void
(*)(VALUE,  VALUE)' {aka 'void (*)(long long unsigned int,  long long unsigned
int)'}
 void rb_define_module_function(VALUE,const char*,VALUE(*)(ANYARGS),int);
                                                  ^~~~~~~~~~~~~~~~~
compiling locker.c
linking shared-object debase_internals.so
debase_internals.o:debase_internals.c:(.text+0x596): undefined reference to
`ibf_load_iseq_complete'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:261: debase_internals.so] Błąd 1

make failed, exit code 2

Gem files will remain installed in
C:/Ruby26-x64/lib/ruby/gems/2.6.0/gems/debase-0.2.3.beta3 for inspection.
Results logged to
C:/Ruby26-x64/lib/ruby/gems/2.6.0/extensions/x64-mingw32/2.6.0/debase-0.2.3.beta3/gem_make.out

An error occurred while installing debase (0.2.3.beta3), and Bundler cannot
continue.
Make sure that `gem install debase -v '0.2.3.beta3' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  debase
```

using https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-2.6.0-1/rubyinstaller-devkit-2.6.0-1-x64.exe

gemfile:
```  
  gem 'debase', '~> 0.2.3.beta3'
  gem 'debase-ruby_core_source', '~> 0.10.4'
```
